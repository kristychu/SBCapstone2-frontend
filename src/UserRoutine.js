import React from 'react';
import { Link } from "react-router-dom";
import { useSelector } from 'react-redux';
import RoutineList from './RoutineList';

function UserRoutine(){
    const token = useSelector(st => st.users.profile.token)
    if(token){
        const decoded = JSON.parse(atob(token.split('.')[1]));
        const username = decoded.username
        return(
            <div>
                <p>Hello {username}!</p>
                <RoutineList/>
            </div>
        )
    }
    return(
        <div>
            <p>You need an account to access this page.</p>
            <p>Have an account already? <Link to="/login">Login!</Link></p>
            <p>Otherwise, <Link to="/register">Register here</Link>.</p>
        </div>
    )
}

export default UserRoutine;