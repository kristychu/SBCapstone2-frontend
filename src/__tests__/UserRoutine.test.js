import React from "react";
import { Provider } from "react-redux";
import { createStore } from "redux";
import reducer from "../reducers/rootReducer";
import { render } from "@testing-library/react";
import UserRoutine from "../UserRoutine";
import { MemoryRouter } from "react-router";

const store = createStore(reducer);

it("should render without crashing", () => {
  render(
    <Provider store={store}>
      <MemoryRouter>
        <UserRoutine />
      </MemoryRouter>
    </Provider>
  );
});

it("should match snapshot", function () {
  const { asFragment } = render(
    <Provider store={store}>
      <MemoryRouter>
        <UserRoutine />
      </MemoryRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});
