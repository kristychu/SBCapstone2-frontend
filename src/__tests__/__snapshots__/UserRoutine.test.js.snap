// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot for anonymous user 1`] = `
<DocumentFragment>
  <div>
    <p>
      You need an account to access this page.
    </p>
    <p>
      Have an account already? 
      <a
        href="/login"
      >
        Login!
      </a>
    </p>
    <p>
      Otherwise, 
      <a
        href="/register"
      >
        Register here
      </a>
      .
    </p>
  </div>
</DocumentFragment>
`;

exports[`should match snapshot if logged in 1`] = `
<DocumentFragment>
  <div>
    <p>
      Hello testuser!
    </p>
    <table
      class="table"
    >
      <thead>
        <tr>
          <th>
            Morning Step
          </th>
          <th>
            You are using:
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Makeup Remover and Oil Cleanser
          </td>
          <td>
            <button
              class="btn btn-info"
              type="button"
            >
              Find Product
            </button>
          </td>
        </tr>
        <tr>
          <td>
            Water Based Cleanser
          </td>
          <td>
            <button
              class="btn btn-info"
              type="button"
            >
              Find Product
            </button>
          </td>
        </tr>
        <tr>
          <td>
            Exfoliator
          </td>
          <td>
            <button
              class="btn btn-info"
              type="button"
            >
              Find Product
            </button>
          </td>
        </tr>
      </tbody>
      <thead>
        <tr>
          <th>
            Night Step
          </th>
          <th>
            You are using:
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Makeup Remover and Oil Cleanser
          </td>
          <td>
            <button
              class="btn btn-info"
              type="button"
            >
              Find Product
            </button>
          </td>
        </tr>
        <tr>
          <td>
            Water Based Cleanser
          </td>
          <td>
            <button
              class="btn btn-info"
              type="button"
            >
              Find Product
            </button>
          </td>
        </tr>
        <tr>
          <td>
            Exfoliator
          </td>
          <td>
            <button
              class="btn btn-info"
              type="button"
            >
              Find Product
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
